plugins/lsp/lsp-lines.nix

```diff
@@ -6,40 +6,36 @@
   ...
 }:
 with lib;
-let
-  cfg = config.plugins.lsp-lines;
-in
-{
-  options = {
-    plugins.lsp-lines = {
-      enable = mkEnableOption "lsp_lines.nvim";
-
-      package = helpers.mkPluginPackageOption "lsp_lines.nvim" pkgs.vimPlugins.lsp_lines-nvim;
+# This plugin has no settings, instead vim's diagnostics should be configured
+helpers.vim-plugin.mkVimPlugin config {
+  name = "lsp-lines";
+  originalName = "lsp_lines.nvim";
+  defaultPackage = pkgs.vimPlugins.lsp_lines-nvim;
 
-      currentLine = mkOption {
-        type = types.bool;
-        default = false;
-        description = "Show diagnostics only on current line";
-      };
-    };
-  };
+  maintainers = [ maintainers.MattSturgeon ];
 
-  config =
-    let
-      diagnosticConfig = {
-        virtual_text = false;
-        virtual_lines = if cfg.currentLine then { only_current_line = true; } else true;
-      };
-    in
-    mkIf cfg.enable {
-      extraPlugins = [ cfg.package ];
+  # TODO: Introduced 2024-06-25, remove after 24.11
+  imports = [
+    (mkRenamedOptionModule
+      [
+        "plugins"
+        "lsp-lines"
+        "currentLine"
+      ]
+      [
+        "diagnostics"
+        "virtual_lines"
+        "only_current_line"
+      ]
+    )
+  ];
 
-      extraConfigLua = ''
-        do
-          require("lsp_lines").setup()
+  extraConfig = cfg: {
+    # Strongly recommended by the plugin, to avoid duplication.
+    diagnostics.virtual_text = mkDefault false;
 
-          vim.diagnostic.config(${helpers.toLuaObject diagnosticConfig})
-        end
-      '';
-    };
+    extraConfigLua = ''
+      require("lsp_lines").setup()
+    '';
```

Comment:
IMO, mkNeovimPlugin should support this usage. Maybe add a `hasSettings ? true` argument which decides whether to create a `settings` option?

Reply:
Yes, I am facing the same issue with md-tui.
I propose to hold this while waiting for the `mk[Neo]VimPlugin` refactoring.

Reply:
Hope I'm not stepping on your toes, but for this specific issue I've made [a small PR](https://github.com/nix-community/nixvim/pull/1758) allowing `settingsOptions` to be set to `null`.

Reply:
This was resolved in https://github.com/nix-community/nixvim/pull/1758